modules/nomad-clients/main.tf

```diff
@@ -80,6 +80,6 @@ resource "aws_security_group_rule" "nomad_client_services" {
   security_group_id = module.nomad_clients.security_group_id
   from_port         = 20000
   to_port           = 32000
-  protocol          = "tcp"
```

Comment:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule

"""
Setting protocol = "all" or protocol = -1 with from_port and to_port will result in the EC2 API creating a security group rule with all ports open. This API behavior cannot be controlled by Terraform and may generate warnings in the future.
"""

I can guess this is so because ICMP protocol is one of the included and has no concept of ports.

Split into two and use "tdp" and "udp" instead.